queue_services/entity-filer/src/entity_filer/filing_processors/incorporation_filing.py

```diff
@@ -126,12 +128,19 @@ def process(
         raise DefaultException(f"Business Already Exist: IA legal_filing:incorporationApplication {filing_rec.id}")
 
     business_info_obj = incorp_filing.get("nameRequest")
+    entity_type = business_info_obj.get("legalType")
+
+    if entity_type in (
+        BusinessCommon.EntityTypes.SOLE_PROP,
+        BusinessCommon.EntityTypes.PARTNERSHIP,
+    ):
+        raise DefaultException(f"Cannot process firm type businesses: IA legal_filing:incorporationApplication {filing_rec.id}")
```

Comment:
don't think we need this?  I think the IA validator in legal api would have checked the legal types already.

Reply:
As far as I can tell from the IA validator, there is nothing preventing firms from reaching the processor. There are some checks on the entity_type in the validator, but it's mainly used for validating parts of the filing json like [share_structure](https://github.com/bcgov/lear/blob/feature-legal-name/legal-api/src/legal_api/services/filings/validations/incorporation_application.py#L73), [incorporator and director counts](https://github.com/bcgov/lear/blob/feature-legal-name/legal-api/src/legal_api/services/filings/validations/incorporation_application.py#L175) etc. 

Probably makes the most sense to move this check to the validator though, right?

Reply:
Yes, you're right, there's no validation for entity_type in both registration and IA filings.  I would probably be hesitant to add it to the validator for now though.  Let's just leave it out if it was never checked in the first place.